vignettes/formats_precedence.Rmd

```diff
@@ -0,0 +1,300 @@
+
+```{r}
+lyt <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  split_rows_by("SEX") %>%
+  analyze(vars = "AGE", afun = mean)
+
+adsl_analyzed <- build_table(lyt, ADSL)
+adsl_analyzed
+table_structure(adsl_analyzed)
+```
+
+For this reporting table, there are 4 sub-tables under `SEX` table, i.e. `F`, `M`, `U`, and `UNDIFFERENTIATED`. Each of these sub-tables has one sub-table `AGE`. So, e.g. for the top sub-table `AGE`, its nearest parent table is `F`.
+
+The concept of formats precedence and inheritance in `rtables` can have multiple levels of parent tables and a final cell level. The cell level is the most specific and lowest level at which the format can be specified by the user. If the cell format is not explicitly specified, it is then inherited from its nearest parent table for which the format was chosen by the user (if any). If no specific format has been selected by the user, then the default format will be applied. The default format is "xx" and it yields the same formatting as `as.character()` function. In summary, the precedence-inheritance chain looks like below.
+```
+cell <- parent_table <- parent_table <- ... <- parent_table
+```
+In case when the format is specified by the user in more than one place, the one which is closer to the `cell` in the precedence-inheritance chain will be applied. In the following sections of this vignette, we illustrate the format precedence rules with a few examples.
+
+### Standard format
+
+Below is a simple layout that does not explicitly set any specific format for the output of the analysis function used in this example. In such a case, the default format will be used.
+
+```{r}
+lyt0 <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(vars = "AGE", afun = mean)
+
+build_table(lyt0, ADSL)
+```
+
+### Cell format
+
+The format of the cell can be explicitly specified through `rcell()` or `in_rows()` functions. The former is essentially a collection of data objects while the latter is a collection of `rcell()` objects. In the context of inheritance, this is the most specific place where the format can be specified by the user.
+
+```{r}
+lyt1 <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(vars = "AGE", afun = function(x) {
+    rcell(mean(x), format = "xx.xx", label = "Mean")
+  })
+
+build_table(lyt1, ADSL)
+
+lyt1a <- basic_table() %>%
+  split_cols_by("ARM") %>%
+  analyze(vars = "AGE", afun = function(x) {
+    in_rows(
+      "Mean" = rcell(mean(x)),
+      .formats = "xx.xx"
+    )
+  })
+
+build_table(lyt1a, ADSL)
+```
+
+If the format is specified in both of these places at the same time, the one specified through `in_rows()`, has higher precedence. Technically, in this case, the format defined in `rcell()` will simply be overwritten by the one defined in `in_rows()`.
```

Comment:
Please explicitly say that this is because the format (and na_str) specified in in_rows is applied to the *cells* not the rows (overriding the previously specified cell specific values), which means that the precedence rules described above are still in place.